DiscordManimator.py

```diff
@@ -1,170 +1,129 @@
-                script=script[6:]
-            elif script.startswith("py"):
-                script=script[2:]
-            
-            script=script.strip()
-
-            if script.startswith("def"):            
-                script="""from manimlib.imports import *
-class test(Scene):
-    """ +"\n    ".join(re.findall('(?:"[^"]*"|.)+|(?!\Z)',script))
-            
-            scenename=str(re.search(r"^class (.*) ?\(.*?$",script,flags=re.M).group(1)).rstrip()
-
-            imports = list_imports.parse(script)
-
-            for package in imports:
-                if package not in ["manimlib.imports","manimlib.constants","numpy","scipy"]:
-                    raise Exception("Your code imports system modules. Can't have that happening!")
-                else:
-                    pass
-
-        except Exception as error:
-            await ctx.send(str(mention)+" Couldn't parse your code... Sorry!" + "\n```"+str(error)+"```")
-            script=None
-
-        if bool(script):
-            try:
-                open("TempManim/temporary"+name+".py","w+").write(script)
-            except Exception as error:
-                await ctx.send(str(mention)+" Couldn't write your code to an internal file... Sorry!") #+"\n```"+str(error)+"```")
+        if arg.startswith('```'): # empty header
+            arg = '\n' + arg
+        header, *body = arg.split('\n')
+
+        cli_flags = header.split()
+        allowed_flags = ["-i", "-s", "-t"]
+        if not all([flag in allowed_flags for flag in cli_flags]):
+            await ctx.reply("You cannot pass CLI flags other than `-i`, `-s`, `-t`.")
+            return
+        else:
+            cli_flags = ' '.join(cli_flags)
+
+        body = '\n'.join(body).strip()
+
+        if not (body.count('```') == 2
+                and body.startswith('```')
+                and body.endswith('```')):
+            await ctx.reply(
+                'Your message is not properly formatted. '
+                'Your code has to be written in a code block, like so:\n'
+                '\\`\\`\\`py\nyour code here\n\\`\\`\\`'
+            )
+            return
+
+        if body.startswith('```python'):
+            script = body[9:-3]
+        elif body.startswith('```py'):
+            script = body[5:-3]
+        else:
+            script = body[3:-3]
+        script = script.strip()
+
+        # for convenience: allow construct-only:
+        if script.startswith('def construct(self):'):
+            script = ['class Manimation(Scene):'] + ["    " + line for line in script.split("\n")]
+        else:
+            script = script.split("\n")
+
+        script = ["from manim import *"] + script
+
+        # write code to temporary file (ideally in temporary directory)
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            scriptfile = Path(tmpdirname) / 'script.py'
+            with open(scriptfile, 'w') as f:
+                f.write('\n'.join(script))
```

Comment:
Space (as in disk space) isn't really an issue anymore (I'll make sure that wherever it will be hosted, this cannot cause any troubles).

I like the `tempfile.TemporaryDirectory` approach quite a lot, because it fully cleans up everything. :-)